packages/graph-engine/src/programmatic/node.ts

```diff
@@ -9,6 +9,18 @@ import { v4 as uuid } from 'uuid';
 import getDefaults from 'json-schema-defaults-esm';
 import type { NodeRun } from '../types.js';
 
+export interface ISubgraphContainer {
+	getSubgraphs(): Graph[];
```

Comment:
You still need to know how to handle the clone here as otherwise you can't handle automatically cloning.  You'd need either extra metadata for the named properties, or to create an actual graph container that can handle the cloning